[WPE][GTK] Fix various -Werror=unsafe-buffer-usage-in-container build failures

mcatanzaro opened the pull request:
#### 769ae115318d950f9c71748e5f381d90176d9874
<pre>
[WPE][GTK] Fix various -Werror=unsafe-buffer-usage-in-container build failures
<a href="https://bugs.webkit.org/show_bug.cgi?id=283684">https://bugs.webkit.org/show_bug.cgi?id=283684</a>

Reviewed by Adrian Perez de Castro.

* Source/WTF/wtf/glib/GSpanExtras.cpp:
(WTF::gFileGetContents):
* Source/WTF/wtf/glib/GSpanExtras.h:
* Source/WebCore/PAL/pal/crypto/gcrypt/CryptoDigestGCrypt.cpp:
(PAL::CryptoDigest::computeHash):
* Source/WebCore/platform/SharedBuffer.cpp:
(WebCore::DataSegment::span const):
* Source/WebCore/platform/SharedBufferChunkReader.cpp:
(WebCore::SharedBufferChunkReader::setSeparator):
* Source/WebCore/platform/Skia.cmake:
* Source/WebCore/platform/WebCorePersistentCoders.cpp:
(WTF::Persistence::Coder&lt;GRefPtr&lt;GByteArray&gt;&gt;::encodeForPersistence):
(WTF::Persistence::Coder&lt;GRefPtr&lt;GByteArray&gt;&gt;::decodeForPersistence):
* Source/WebCore/platform/audio/glib/AudioBusGLib.cpp:
(WebCore::AudioBus::loadPlatformResource):
* Source/WebCore/platform/audio/gstreamer/AudioDecoderGStreamer.cpp:
(WebCore::GStreamerAudioDecoder::decode):
* Source/WebCore/platform/audio/gstreamer/PlatformRawAudioDataGStreamer.cpp:
(WebCore::PlatformRawAudioData::copyTo):
* Source/WebCore/platform/glib/KeyedDecoderGlib.cpp:
(WebCore::KeyedDecoderGlib::decodeBytes):
* Source/WebCore/platform/glib/KeyedEncoderGlib.cpp:
(WebCore::KeyedEncoderGlib::finishEncoding):
* Source/WebCore/platform/graphics/WOFFFileFormat.cpp:
(WebCore::readUInt32):
(WebCore::readUInt16):
(WebCore::writeUInt32):
(WebCore::writeUInt16):
* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
(WebCore::extractGStreamerOptionsFromCommandLine):
(WebCore::GstMappedBuffer::createVector const):
* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.h:
(WebCore::unmapFunction):
* Source/WebCore/platform/graphics/gstreamer/InbandTextTrackPrivateGStreamer.cpp:
(WebCore::InbandTextTrackPrivateGStreamer::notifyTrackOfSample):
* Source/WebCore/platform/graphics/gstreamer/TrackPrivateBaseGStreamer.cpp:
(WebCore::TrackPrivateBaseGStreamer::updateTrackIDFromTags):
* Source/WebCore/platform/graphics/gstreamer/VideoDecoderGStreamer.cpp:
(WebCore::GStreamerVideoDecoder::decode):
* Source/WebCore/platform/graphics/gtk/ImageAdapterGtk.cpp:
(WebCore::loadImageFromGResource):
* Source/WebCore/platform/graphics/skia/SkiaSpanExtras.h: Added.
(WebCore::span):
* Source/WebCore/platform/gstreamer/GStreamerQuirks.cpp:
(WebCore::GStreamerQuirksManager::GStreamerQuirksManager):
* Source/WebCore/platform/gtk/PlatformKeyboardEventGtk.cpp:
(WebCore::PlatformKeyboardEvent::singleCharacterString):
* Source/WebCore/platform/image-decoders/ico/ICOImageDecoder.cpp:
(WebCore::ICOImageDecoder::imageTypeAtIndex):
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateGStreamer.cpp:
(WebCore::MediaRecorderPrivateBackend::processSample):
* Source/WebCore/platform/network/soup/NetworkStorageSessionSoup.cpp:
(WebCore::NetworkStorageSession::getCredentialFromPersistentStorage):
* Source/WebCore/platform/network/soup/SoupNetworkSession.cpp:
(WebCore::HostTLSCertificateSet::computeCertificateHash):
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::mediaControlsBase64StringForIconNameAndType):
* Source/WebKit/Shared/skia/CoreIPCSkString.h:
(WebKit::CoreIPCSkString::data const):
* Source/WebKit/UIProcess/API/glib/WebKitWebResource.cpp:
(resourceDataCallback):

Canonical link: <a href="https://commits.webkit.org/287621@main">https://commits.webkit.org/287621@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c7463e93b87b629f39593d3b4d0ebf1bade672bc

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/80337 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/48/builds/59343 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/33818 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/84858 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/31319 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/82448 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/68405 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/123/builds/7644 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/84858 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/31319 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/83406 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/52868 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/73144 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/84858 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/50187 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/27299 "Passed tests") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/29778 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/73317 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/71326 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/27813 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/86292 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/79396 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/7562 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/123/builds/7644 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/86292 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/7737 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/68980 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/86292 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/13251 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/101803 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12421 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/7524 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/13044 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/24809 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/7363 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/10883 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/9169 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebCore/platform/WebCorePersistentCoders.cpp:
```diff
@@ -446,8 +450,13 @@ template<> struct Coder<GRefPtr<GByteArray>> {
+        WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // WPE/GTK ports
+        // FIXME: Can't use WTF::span here because it returns std::span<const uint8_t>
+        // but we need std::span<uint8_t>.
         if (!decoder.decodeFixedLengthData({ byteArray->data, *size }))
```
You can use `WTF::spanConstCast`, I think this should do:

```suggestion
        if (!decoder.decodeFixedLengthData(WTF::spanConstCast<uint8_t>(WTF::span(byteArray))))
```

mcatanzaro replied:
Nice.

aperezdc commented on Source/WebCore/platform/SharedBuffer.cpp:
```diff
@@ -620,13 +624,13 @@ std::span<const uint8_t> DataSegment::span() const
 #endif
 #if USE(SKIA)
-        [](const sk_sp<SkData>& data) -> std::span<const uint8_t> { return { data->bytes(), data->size() }; },
+        [](const sk_sp<SkData>& data) -> std::span<const uint8_t> { return unsafeMakeSpan(data->bytes(), data->size()); },
```
I would add a new `span(const sk_sp<SkData>&) -> std::span<const uint8_t>` helper function and use it here. It would have to go in WebCore because I suppose we do not want to make WTF depend on Skia 😉. Then that helper function could be used in other places (e.g. in `CoreIPCSkData.h`, `CoreIPCSkColorSpace.h`, and probable a few more).

mcatanzaro replied:
Hi, any suggestions for where we could put this? Perhaps WebCore/platform/graphics/SkSpanExtras.h?

mcatanzaro replied:
I decided to use WebCore/platform/graphics/SkiaSpanExtras.h unless you have a better idea.

aperezdc commented on Source/WebCore/platform/audio/gstreamer/PlatformRawAudioDataGStreamer.cpp:
```diff
@@ -225,7 +225,7 @@ void PlatformRawAudioData::copyTo(std::span<uint8_t> destination, AudioSampleFor
     GUniquePtr<GstAudioInfo> sourceInfo(gst_audio_info_copy(self.info()));
     GUniquePtr<char> key(gst_info_strdup_printf("%" GST_PTR_FORMAT ";%" GST_PTR_FORMAT, gst_sample_get_caps(self.sample()), outputCaps.get()));
-    auto converter = getAudioConvertedForFormat(StringView { std::span { key.get(), strlen(key.get()) } }, *sourceInfo.get(), destinationInfo);
+    auto converter = getAudioConvertedForFormat(StringView { unsafeMakeSpan(key.get(), strlen(key.get())) }, *sourceInfo.get(), destinationInfo);
```
Maybe we could use `StringView::fromLatin1(key.get())` directly. Wouldn't `gst_info_strdup_printf()` return always ASCII?

cdumez replied:
You can just call `span(key.get())` from StringCommon.h.

aperezdc commented on Source/WebCore/platform/graphics/WOFFFileFormat.cpp:
```diff
@@ -72,13 +72,13 @@ static bool readUInt16(SharedBuffer& buffer, size_t& offset, uint16_t& value)
 {
     uint32_t bigEndianValue = htonl(value);
-    return vector.tryAppend(std::span { reinterpret_cast_ptr<uint8_t*>(&bigEndianValue), sizeof(bigEndianValue) });
+    return vector.tryAppend(unsafeMakeSpan(reinterpret_cast_ptr<uint8_t*>(&bigEndianValue), sizeof(bigEndianValue)));
```
There's `WTF::asByteSpan(const T&)`:

```suggestion
    return vector.tryAppend(asByteSpan(bigEndianValue));
```

aperezdc commented on Source/WebCore/platform/graphics/WOFFFileFormat.cpp:
```diff
@@ -72,13 +72,13 @@ static bool readUInt16(SharedBuffer& buffer, size_t& offset, uint16_t& value)
 {
     uint16_t bigEndianValue = htons(value);
-    return vector.tryAppend(std::span { reinterpret_cast_ptr<uint8_t*>(&bigEndianValue), sizeof(bigEndianValue) });
+    return vector.tryAppend(unsafeMakeSpan(reinterpret_cast_ptr<uint8_t*>(&bigEndianValue), sizeof(bigEndianValue)));
```
```suggestion
    return vector.tryAppend(asByteSpan(bigEndianValue));
```

aperezdc commented on Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
```diff
@@ -910,7 +910,7 @@ void connectSimpleBusMessageCallback(GstElement* pipeline, Function<void(GstMess
 Vector<uint8_t> GstMappedBuffer::createVector() const
 {
-    return std::span<const uint8_t> { data(), size() };
+    return unsafeMakeSpan<const uint8_t>(data(), size());
```
There's a `GstBufferMapper::span()` function, which is inherited by `GstMappedBuffer`:

```suggestion
    return Vector(span());
```

aperezdc commented on Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
```diff
@@ -340,7 +340,7 @@ Vector<String> extractGStreamerOptionsFromCommandLine()
 
     Vector<String> options;
-    auto optionsString = String::fromUTF8(std::span(contents.get(), length));
+    auto optionsString = String::fromUTF8(unsafeMakeSpan(contents.get(), length));
```
Probably it would be a good idea to add this to `GSpanExtras.h`:

```cpp
GMallocSpan<char> gFileGetContents(const char* path, GUniqueOutPtr<GError>&);
```

and then use it here and in the few other places where we have calls to `g_file_get_contents()`.

mcatanzaro replied:
I've added the helper, but the only other place where we could use it is ChassisType.cpp, and I don't think it actually helps there.

aperezdc commented on Source/WebCore/platform/graphics/gstreamer/InbandTextTrackPrivateGStreamer.cpp:
```diff
@@ -131,7 +131,7 @@ void InbandTextTrackPrivateGStreamer::notifyTrackOfSample()
         ASSERT(!hasClients() || hasOneClient());
         notifyMainThreadClient([&](auto& client) {
-            downcast<InbandTextTrackPrivateClient>(client).parseWebVTTCueData(std::span { mappedBuffer.data(), mappedBuffer.size() });
+            downcast<InbandTextTrackPrivateClient>(client).parseWebVTTCueData(unsafeMakeSpan(mappedBuffer.data(), mappedBuffer.size()));
```
Same as above, there is `GstBufferMapper::span()`:

```suggestion
            downcast<InbandTextTrackPrivateClient>(client).parseWebVTTCueData(mappedBuffer.span());
```

aperezdc commented on Source/WebCore/platform/graphics/gstreamer/TrackPrivateBaseGStreamer.cpp:
```diff
@@ -372,7 +372,7 @@ bool TrackPrivateBaseGStreamer::updateTrackIDFromTags(const GRefPtr<GstTagList>&
         return false;
 
-    auto trackID = WTF::parseInteger<TrackID>(StringView { std::span { trackIDString.get(), strlen(trackIDString.get()) } });
+    auto trackID = WTF::parseInteger<TrackID>(StringView { unsafeMakeSpan(trackIDString.get(), strlen(trackIDString.get())) });
```
It would be nice to know if track identifiers will always be ASCII, and use `StringView::fromLatin1()` in that case.

cdumez replied:
Can call `span(trackIDString.get())`

aperezdc commented on Source/WebCore/platform/gstreamer/GStreamerQuirks.cpp:
```diff
@@ -76,7 +76,7 @@ GStreamerQuirksManager::GStreamerQuirksManager(bool isForTesting, bool loadQuirk
     GST_DEBUG("Attempting to parse requested quirks: %s", GST_STR_NULL(quirksList));
     if (quirksList) {
-        StringView quirks { std::span { quirksList, strlen(quirksList) } };
+        StringView quirks { unsafeMakeSpan(quirksList, strlen(quirksList)) };
```
I'm pretty sure this should be ASCII-only, we could use `StringView::fromLatin1()` here.

cdumez replied:
Can call `span(quirksList)`

aperezdc commented on Source/WebCore/platform/gstreamer/GStreamerQuirks.cpp:
```diff
@@ -114,7 +114,7 @@ GStreamerQuirksManager::GStreamerQuirksManager(bool isForTesting, bool loadQuirk
         return;
 
-    StringView identifier { std::span { holePunchQuirk, strlen(holePunchQuirk) } };
+    StringView identifier { unsafeMakeSpan(holePunchQuirk, strlen(holePunchQuirk)) };
```
Same here.

cdumez replied:
Can call `span(holePunchQuirk)`

aperezdc commented on Source/WebCore/platform/image-decoders/ico/ICOImageDecoder.cpp:
```diff
@@ -321,7 +321,7 @@ ICOImageDecoder::ImageType ICOImageDecoder::imageTypeAtIndex(size_t index)
     if ((imageOffset > m_data->size()) || ((m_data->size() - imageOffset) < 4))
         return Unknown;
-    return equalSpans(m_data->span().subspan(imageOffset, 4), std::span { "\x89PNG", 4 }) ? PNG : BMP;
+    return equalSpans(m_data->span().subspan(imageOffset, 4), unsafeMakeSpan("\x89PNG", 4)) ? PNG : BMP;
```
Any reason to not use a `""_span` literal?

```suggestion
    return equalSpans(m_data->span().subspan(imageOffset, 4), "\x89PNG"_span) ? PNG : BMP;
```

aperezdc commented on Source/WebCore/platform/mediarecorder/MediaRecorderPrivateGStreamer.cpp:
```diff
@@ -476,7 +476,7 @@ void MediaRecorderPrivateBackend::processSample(GRefPtr<GstSample>&& sample)
 
     GST_LOG_OBJECT(m_transcoder.get(), "Queueing %zu bytes of encoded data, caps: %" GST_PTR_FORMAT, buffer.size(), gst_sample_get_caps(sample.get()));
-    m_data.append(std::span<const uint8_t> { buffer.data(), buffer.size() });
+    m_data.append(unsafeMakeSpan<const uint8_t>(buffer.data(), buffer.size()));
```
Same as above:

```suggestion
    m_data.append(buffer.span());
```

aperezdc commented on Source/WebKit/UIProcess/API/glib/WebKitWebResource.cpp:
```diff
@@ -362,7 +362,7 @@ static void resourceDataCallback(API::Data* wkData, GTask* task)
     data->webData = wkData;
     if (!wkData->span().data())
-        data->webData = API::Data::create({ reinterpret_cast<const uint8_t*>(""), 1 });
+        data->webData = API::Data::create(unsafeMakeSpan(reinterpret_cast<const uint8_t*>(""), 1));
```
```suggestion
        data->webData = API::Data::create(""_span);
```

mcatanzaro replied:
EWS doesn't like this change so I'll omit this for now.

aperezdc reviewed (changes requested):
Thanks for working on this! I left comments 😉

mcatanzaro commented on Source/WebCore/platform/SharedBufferChunkReader.cpp:
```diff
@@ -62,7 +64,7 @@ void SharedBufferChunkReader::setSeparator(const Vector<char>& separator)
 {
     m_separator.clear();
-    m_separator.append(span(separator));
+    m_separator.append(WTF::span(separator));
```
(Note: without the explicit use of WTF namespace here, Clang tries to use the new `WebCore::span` instead and complains that it cannot cast to the Skia type.)

philn commented on Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.h:
```diff
@@ -455,3 +456,4 @@ GstBuffer* gst_buffer_new_memdup(gconstpointer data, gsize size);
 #endif
 
 #endif // USE(GSTREAMER)
+
```
Nit: maybe remove this empty line

mcatanzaro replied:
Will do before landing. I think Builder is inserting it. <_<

philn reviewed (commented):
The gst-related changes lgtm, thanks!

aperezdc reviewed (approved):
Definitely nicer with the last round of changes, thanks! r=me

mcatanzaro committed:
[WPE][GTK] Fix various -Werror=unsafe-buffer-usage-in-container build failures
https://bugs.webkit.org/show_bug.cgi?id=283684

Reviewed by Adrian Perez de Castro.

* Source/WTF/wtf/glib/GSpanExtras.cpp:
(WTF::gFileGetContents):
* Source/WTF/wtf/glib/GSpanExtras.h:
* Source/WebCore/PAL/pal/crypto/gcrypt/CryptoDigestGCrypt.cpp:
(PAL::CryptoDigest::computeHash):
* Source/WebCore/platform/SharedBuffer.cpp:
(WebCore::DataSegment::span const):
* Source/WebCore/platform/SharedBufferChunkReader.cpp:
(WebCore::SharedBufferChunkReader::setSeparator):
* Source/WebCore/platform/Skia.cmake:
* Source/WebCore/platform/WebCorePersistentCoders.cpp:
(WTF::Persistence::Coder<GRefPtr<GByteArray>>::encodeForPersistence):
(WTF::Persistence::Coder<GRefPtr<GByteArray>>::decodeForPersistence):
* Source/WebCore/platform/audio/glib/AudioBusGLib.cpp:
(WebCore::AudioBus::loadPlatformResource):
* Source/WebCore/platform/audio/gstreamer/AudioDecoderGStreamer.cpp:
(WebCore::GStreamerAudioDecoder::decode):
* Source/WebCore/platform/audio/gstreamer/PlatformRawAudioDataGStreamer.cpp:
(WebCore::PlatformRawAudioData::copyTo):
* Source/WebCore/platform/glib/KeyedDecoderGlib.cpp:
(WebCore::KeyedDecoderGlib::decodeBytes):
* Source/WebCore/platform/glib/KeyedEncoderGlib.cpp:
(WebCore::KeyedEncoderGlib::finishEncoding):
* Source/WebCore/platform/graphics/WOFFFileFormat.cpp:
(WebCore::readUInt32):
(WebCore::readUInt16):
(WebCore::writeUInt32):
(WebCore::writeUInt16):
* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
(WebCore::extractGStreamerOptionsFromCommandLine):
(WebCore::GstMappedBuffer::createVector const):
* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.h:
(WebCore::unmapFunction):
* Source/WebCore/platform/graphics/gstreamer/InbandTextTrackPrivateGStreamer.cpp:
(WebCore::InbandTextTrackPrivateGStreamer::notifyTrackOfSample):
* Source/WebCore/platform/graphics/gstreamer/TrackPrivateBaseGStreamer.cpp:
(WebCore::TrackPrivateBaseGStreamer::updateTrackIDFromTags):
* Source/WebCore/platform/graphics/gstreamer/VideoDecoderGStreamer.cpp:
(WebCore::GStreamerVideoDecoder::decode):
* Source/WebCore/platform/graphics/gtk/ImageAdapterGtk.cpp:
(WebCore::loadImageFromGResource):
* Source/WebCore/platform/graphics/skia/SkiaSpanExtras.h: Added.
(WebCore::span):
* Source/WebCore/platform/gstreamer/GStreamerQuirks.cpp:
(WebCore::GStreamerQuirksManager::GStreamerQuirksManager):
* Source/WebCore/platform/gtk/PlatformKeyboardEventGtk.cpp:
(WebCore::PlatformKeyboardEvent::singleCharacterString):
* Source/WebCore/platform/image-decoders/ico/ICOImageDecoder.cpp:
(WebCore::ICOImageDecoder::imageTypeAtIndex):
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateGStreamer.cpp:
(WebCore::MediaRecorderPrivateBackend::processSample):
* Source/WebCore/platform/network/soup/NetworkStorageSessionSoup.cpp:
(WebCore::NetworkStorageSession::getCredentialFromPersistentStorage):
* Source/WebCore/platform/network/soup/SoupNetworkSession.cpp:
(WebCore::HostTLSCertificateSet::computeCertificateHash):
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::mediaControlsBase64StringForIconNameAndType):
* Source/WebKit/Shared/skia/CoreIPCSkString.h:
(WebKit::CoreIPCSkString::data const):
* Source/WebKit/UIProcess/API/glib/WebKitWebResource.cpp:
(resourceDataCallback):

Canonical link: https://commits.webkit.org/287621@main
